<launch>
  <arg name="voxel_size" default="0.01"/>

  <!-- Download the dataset here: https://projects.asl.ethz.ch/datasets/doku.php?id=iros2017 -->

  <node name="voxblox_node" pkg="map_adapt_ros" type="adaptive_voxel_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="/traj_visualizer/pcl_output"/>
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <!-- <param name="truncation_distance" value="0.02" /> -->
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="false" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="false" />
    <param name="update_mesh_every_n_sec" value="2" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="method" value="fast" />
    <param name="use_const_weight" value="simple" />
    <param name="allow_clear" value="false" />
    <param name="verbose" value="true" />
    <param name="min_ray_length_m" value="0.0" />
    <param name="max_ray_length_m" value="8.0" />
    <param name="visualize_semantic" value="false" />
    <param name="max_consecutive_ray_collisions" value="3" />
    <param name="load_semantic_probability" value="true" />
    <param name="theta_alpha" value="0.1" />
    <param name="theta_beta" value="40" />
    <param name="theta_gamma" value="0.2" />
    <param name="crf_threads" value="8" />
    <param name="crf_type" value="appearance_and_smoothness" /> <!--straight_forward/appearance/appearance_and_smoothness-->
    <param name="crf_only_on_updated_voxels" value="false" />
    <param name="weight_appearance" value="10" />
    <param name="weight_smooth" value="3" /> <!--3-->
    <param name="adaptive_mapping" value="false" />
    <param name="adaptive_ratio_small" value="8.0" />
    <param name="adaptive_ratio_middle" value="2.0" />
    <param name="fuse_semantic_for_subvoxel" value="false" />
    <param name="intensity_colormap" value="nyu" />
    <param name="intensity_max_value" value="40.0" />
    <param name="semantic_update_method" value="bayesian" /> <!--Support: max_pooling; weighted_max; bayesian; bayesian_constant-->
    <!--collection1: 4(bed),9(window),11(picture); collection2: 6(sofa); collection3: 6(sofa),7(table)-->
    <rosparam param="small_semantic" >[4,9,11]</rosparam> <!--Note: label starts from 1!!!!!,,  collection1: 4,9;collection3: 11-->
    <rosparam param="large_semantic" >[1,2,22]</rosparam> <!--Note: label starts from 1!!!!!-->
    <!-- [1,2,22,0,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40] -->
    <remap from="transform" to="/traj_visualizer/tf_output" />
    <rosparam file="$(find map_adapt_ros)/cfg/identity.yaml"/>
    <param name="mesh_filename" value="$(find map_adapt_ros)/mesh_results/jianhao_habitat.ply" />
  </node>

</launch>
